components/scream/src/share/field/field_repository.hpp

```diff
@@ -307,11 +388,193 @@ void FieldRepository<RealType>::registration_begins () {
 }
 
 template<typename RealType>
-void FieldRepository<RealType>::registration_ends () {
-  // Proceed to allocate fields
-  for (auto& map : m_fields) {
-    for (auto& it : map.second) {
-      it.second.allocate_view();
+void FieldRepository<RealType>::
+registration_ends (const std::shared_ptr<const GridsManager>& gm) {
```

Comment:
Are we requiring a grids manager here because it may not have been created before registration began?

Reply:
I originally modified the ctor of FieldRepository, so that one would have to pass a GM there. But I changed my mind, because it is only used in this fcn, so there's really no need for this class to _store_ a copy.

The GM is only needed to create a vector layout for the given grid, so that tracers can be allocated monolithically as a "vector" quantity.

I'll add a comment near the fcn declaration.